src/Workspaces/Remote/Core/Services/SolutionCreator.cs

```diff
@@ -497,5 +507,47 @@ private Project AddDocument(Project project, DocumentInfo documentInfo)
         {
             return project.Solution.AddDocument(documentInfo).GetProject(project.Id);
         }
+
+        private CompilationOptions FixUpCompilationOptions(ProjectInfo.ProjectAttributes info, CompilationOptions compilationOptions)
+        {
+            // see AbstractProject.CreateCompilationOptions for more detail
+            var projectDirectory = GetContainingDirectoryPathOpt(info.FilePath);
+
+            return compilationOptions.WithXmlReferenceResolver(new XmlFileResolver(projectDirectory))
+                                     .WithStrongNameProvider(new DesktopStrongNameProvider(GetStrongNameKeyPaths(info)));
+        }
+
+        private ImmutableArray<string> GetStrongNameKeyPaths(ProjectInfo.ProjectAttributes info)
+        {
+            if (info.FilePath == null && info.OutputFilePath == null)
+            {
+                return ImmutableArray<string>.Empty;
```

Comment:
Looks like the default parameter value for DesktopStrongNameProvider is actually ```default(ImmutableArray<string>)```.  It might be more obviously correct to pass that instead of empty.